enumerable_methods.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |item|
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result.push(item) if yield(item) }
+    result
+  end
+
+  def my_all?(args = nil)
+    condition = true
+    if block_given?
+      my_each { |item| condition = false if yield(item) }
+    elsif args.nil?
+      my_each { |item| condition = false if item == true }
+    else
+      my_each { |item| condition = false if item === args }
+    end
+    condition
+  end
+
+  def my_any?(args = nil)
+    condition = false
+    if args.nil?
+      my_each { |item| condition = true if yield(item) }
```

Comment:
This part is not working. Your method is returning an error when no block is given.
`[1,2,3, 'string', 'another', 'text'].my_any?`